sdk/core/core-rest-pipeline/test/throttlingRetryPolicy.spec.ts

```diff
@@ -216,4 +217,39 @@ describe("throttlingRetryPolicy", function () {
 
     clock.restore();
   });
+
+  it("throttlingRetryPolicy should honor abort signal", async () => {
+    const request = createPipelineRequest({
+      url: "https://bing.com",
+      abortSignal: AbortController.timeout(100), // test should end at 100ms
+    });
+    const retryResponse: PipelineResponse = {
+      headers: createHttpHeaders({
+        "Retry-After": "10000", // 10000 seconds - a large duration
+      }),
+      request,
+      status: 429,
+    };
+    const successResponse: PipelineResponse = {
+      headers: createHttpHeaders(),
+      request,
+      status: 200,
+    };
+
+    const policy = throttlingRetryPolicy();
+    const next = sinon.stub<Parameters<SendRequest>, ReturnType<SendRequest>>();
+    next.onFirstCall().resolves(retryResponse);
+    next.onSecondCall().resolves(successResponse);
+
+    let errorWasThrown = false;
+    try {
+      await policy.sendRequest(request, next);
+    } catch (error) {
+      errorWasThrown = true;
+      assert.equal((error as Error).name, "AbortError", "Unexpected error thrown");
+    }
+    assert.equal(errorWasThrown, true, "Error was not thrown");
```

Comment:
nit: this could be rewritten more concisely:
```suggestion
    await assert.fail(() => policy.sendRequest(request, next), /<error message>/);
```

Reply:
Cool, can I check the error name with this?